cpp/cmake/thirdparty/get_arrow.cmake

```diff
@@ -34,20 +34,34 @@ function(find_libarrow_in_python_wheel PYARROW_VERSION)
   # version number soname, just `${MAJOR_VERSION}00`
   set(PYARROW_LIB "libarrow.so.${PYARROW_SO_VER}00")
 
-  find_package(Python REQUIRED)
```

Comment:
We should check to make sure `Python_EXECUTABLE` is defined and error out otherwise. In theory, a single scope could have included `get_arrow.cmake` and a different scope calls `find_and_configure_arrow` so we end up with an invalid `execute_process`